Duplicati/Library/Main/Operation/DeleteHandler.cs

```diff
@@ -221,34 +228,41 @@ private DateTime[] GetFilesetsToDelete(Database.LocalDeleteDatabase db, DateTime
             var fullVersionsKeptCount = 0;
             if (fullVersionsToKeep > 0 && fullVersionsToKeep < backupsRemaining.Count)

```

Comment:
Does this make sense? If fullVersionsToKeep is like 50 and backupsRemaining is 3, with 1 full, 1 partial and one full, then we still want to clean the partial in between I guess? Then we should still enter the clean up logic within the if.

Reply:
My thinking was that if there are fewer backups remaining than the number we want to keep, we might as well keep them all (assuming that storage space is the motivation for the limit).  The partial "might" still be useful in some cases.  I think the existence of partials should be relatively rare, so it doesn't hurt to be conservative and not delete them unless we can make a strong case for doing so.